k8s/k8s.go

```diff
@@ -316,6 +445,29 @@ func overrideIPAMResult(ipAddrsNoIpam string, logger *log.Entry) (*types.Result,
 	return &result, nil
 }
 
+// parseIPAddrs is a utility function that parses string of IPs in json format that are
+// passed in as a string and returns a slice of string with IPs.
+// It also makes sure the slice isn't empty.
+func parseIPAddrs(annotation, ipAddrsStr string, logger *log.Entry) ([]string, error) {
```

Comment:
hmm I don't know if I agree, there are a few places we log info/failure inside the function, and having the annotation there would help a lot. 
Like: `logger.WithField("annotation", annotation).Errorf("Failed to parse '%s' as JSON", annotation)` knowing which annotation this was helps a lot. 

Maybe im misunderstanding what you're saying